sandbox/libs/authn/src/main/java/org/opensearch/authn/AuthenticationTokenHandler.java

```diff
@@ -73,4 +80,10 @@ private static AuthenticationToken handleBasicAuth(final BasicAuthToken token) {
 
         return new UsernamePasswordToken(username, password);
     }
+
+    private static AuthenticationToken handleBearerAuth(final BearerAuthToken token) throws BadCredentialsException {
```

Comment:
This will never throw any exception except NPE if `token` is null. So we can remove this from signature.

Reply:
I will check on these exception throws and the try/catch!

Reply:
See the below comment for discussion of handling extractShiroToken. I am not sure what alternative you had in mind should we remove the throw decleration entirely from the extractShiroToken method--I did remove the dead throw of course (great catch)